torch/csrc/jit/codegen/cuda/expr_simplifier.cpp

```diff
@@ -711,19 +944,82 @@ namespace prove {
 // - x can be either zero or non-zero, it is just a symbolic number that depends
 // - x is zero
 
-bool isNonZero(Val* value) {
+bool isNonNegative(Val* value) {
   value = foldConstants(value);
-  if (value->getInt().has_value() && *value->getInt() != 0) {
+  if (value->getInt().has_value() && *value->getInt() >= 0) {
     return true;
   }
-  if (value->getDouble().has_value() && *value->getDouble() != 0.0) {
+  if (value->getDouble().has_value() && *value->getDouble() >= 0.0) {
+    return true;
+  }
+  if (auto ns = dynamic_cast<NamedScalar*>(value)) {
+    if (ns->getParallelDim().has_value() ||
+        ns->getParallelIndex().has_value()) {
+      return true;
+    }
+  }
+  if (auto fop = dynamic_cast<FOp*>(value->definition())) {
+    auto op = fop->getOpType();
+    if (op == BinaryOpType::Add || op == BinaryOpType::Mul) {
+      for (auto inp : fop->inputs()) {
+        if (!isNonNegative(inp)) {
+          return false;
+        }
+      }
+      return true;
+    }
+  }
+  return false;
+}
+
+bool isPositive(Val* value) {
+  value = foldConstants(value);
+  if (value->getInt().has_value() && *value->getInt() > 0) {
+    return true;
+  }
+  if (value->getDouble().has_value() && *value->getDouble() > 0.0) {
     return true;
   }
   if (auto ns = dynamic_cast<NamedScalar*>(value)) {
     if (ns->getParallelDim().has_value()) {
       return true;
     }
   }
+  if (auto fop = dynamic_cast<FOp*>(value->definition())) {
+    auto op = fop->getOpType();
+    if (op == BinaryOpType::Add) {
+      bool has_positive = false;
+      for (auto inp : fop->inputs()) {
+        if (isPositive(inp)) {
+          has_positive = true;
+        } else if (!isNonNegative(inp)) {
+          return false;
+        }
+      }
+      return has_positive;
+    } else if (op == BinaryOpType::Mul) {
+      for (auto inp : fop->inputs()) {
+        if (!isPositive(inp)) {
+          return false;
+        }
+      }
+      return true;
+    }
+  }
+  return false;
+}
+
+bool isNonZero(Val* value) {
```

Comment:
No need to read this, it is just a temporary solution allowing me to create non-zero variables in order to trigger `simplifyDivisibleDivMod` in unit tests. This will be replaced by https://github.com/csarofeen/pytorch/pull/2273